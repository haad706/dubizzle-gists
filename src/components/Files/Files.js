import styled from "styled-components";
import { IconText } from "components";
import { isEmpty } from "utils/helper";
export const Files = ({ files }) => {
  /**
   * incase of no files
   */
  if (isEmpty(files)) {
    return <p>No files found</p>;
  }

  return (
    <FilesWrapper>
      {files.map(({ filename, raw_url: rawUrl }) => (
        <IconText
          iconName="file"
          link={rawUrl}
          text={filename}
          key={filename}
        />
      ))}
    </FilesWrapper>
  );
};

const FilesWrapper = styled.div`
  display: inline-flex;
  gap: 0.5rem;
  align-items: center;
  font-weight: 500;
  flex-wrap: wrap;
`;

export default Files;
