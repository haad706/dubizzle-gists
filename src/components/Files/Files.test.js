import { render } from "@testing-library/react";
import Files from "./Files";

const files = [
  {
    filename: "gistfile1.txt",
    raw_url:
      "https://gist.githubusercontent.com/dwfault/80059b65e85973dbd629dff8d041588d/raw/3599c7641b180f2a4b6041d5e0f2e38f073f0fe6/gistfile1.txt",
  },
  {
    filename: "test.xml",
    raw_url:
      "https://gist.githubusercontent.com/Bezarius/cf8d6d227abb4450fa548c15148d5285/raw/b8aad7d46618739da077c0e0a84d26f805f2f929/test.xml",
  },
];

describe("Files", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<Files files={files} />);
    expect(baseElement).toBeTruthy();
  });
  it("should match snapshot", () => {
    const { asFragment } = render(<Files files={files} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
