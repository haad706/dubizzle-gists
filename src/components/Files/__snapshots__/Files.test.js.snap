// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Files should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jeCNp gPCBRI"
  >
    <div
      class="sc-ispOId iBwrDM"
    >
      <span
        class="octicon octicon-file"
      />
      <a
        class="sc-kcuKUB IWIsi"
        href="https://gist.githubusercontent.com/dwfault/80059b65e85973dbd629dff8d041588d/raw/3599c7641b180f2a4b6041d5e0f2e38f073f0fe6/gistfile1.txt"
        target="_blank"
      >
        gistfile1.txt
      </a>
    </div>
    <div
      class="sc-ispOId iBwrDM"
    >
      <span
        class="octicon octicon-file"
      />
      <a
        class="sc-kcuKUB IWIsi"
        href="https://gist.githubusercontent.com/Bezarius/cf8d6d227abb4450fa548c15148d5285/raw/b8aad7d46618739da077c0e0a84d26f805f2f929/test.xml"
        target="_blank"
      >
        test.xml
      </a>
    </div>
  </div>
</DocumentFragment>
`;
